packages/web-components/fast-components/src/design-system-provider/design-system-provider.ts

```diff
@@ -182,8 +184,28 @@ export class DesignSystemProvider extends FASTElement
                 });
             }
         });
+
+        // Register all properties that may have been attached before construction
+        if (Array.isArray(this.disconnectedCSSCustomPropertyRegistry)) {
+            for (let i = 0; i < this.disconnectedCSSCustomPropertyRegistry.length; i++) {
```

Comment:
nit: you could store the length of this

Reply:
it's actually not faster! https://jsperf.com/loops/33

Reply:
Length cache with pre-increment was actually fastest for me. It may depend on the browser.

Reply:
strange, runs slightly faster for me, but it's really negligible.